// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

// option java_multiple_files = true;
// option java_package = "io.grpc.examples.routeguide";
// option java_outer_classname = "ChatServiceProto";
// option objc_class_prefix = "CS";

package chatservice;

// Interface exported by the server.
service ChatServicer {
  rpc CreateAccount(Username) returns (ChatResponse);

  rpc DeleteAccount(Username) returns (ChatResponse);

  rpc ListAccounts(Wildcard) returns (ListofAccounts);

  rpc SendMessage(Message) returns (ChatResponse);

  rpc DeliverMessages(Username) returns (ChatResponse);
}

message Username {
  string name = 1;
}

message ChatResponse {
  string reply = 1;
}

message ListofAccounts {
  repeated Username usernames=1;
}

message Wildcard {
  string wildcard = 1;
}

message Message {
  string message = 1;
}
